### PR TITLE
Update + Fix MosaicML Composer integration docs

### PR DESCRIPTION
**Pull request by morganmcg1**
Updated the MosaicML docs:
- fixed broken colab link
- fixed 404s linking to old MosaicML Composer docs
- Updated arguments table to new arguments in Composer wandblogger
- updated code snippets

# Testing
Ran it locally and all looks good:

Screenshot 2024-01-05 at 14 36 01

Screenshot 2024-01-05 at 14 36 06

Screenshot 2024-01-05 at 14 36 13

### CODE REVIEW

**Review by ngrayluna — APPROVED** [review]
Approved. QQ: @morganmcg1 I see screen shots for the page, but the PR itself is a one line change. Are there other commits you need to push?

**Comment by morganmcg1** [comment]
jeeez... haha thanks for the catch @ngrayluna, pushed now

**Comment by ngrayluna** [comment]
@morganmcg1 Thanks for pushing the changes. Can you take a look at the comments generated by the doc lint checker?

File: docs/guides/integrations/other/composer.md [hunk 1]
```diff
@@ -6,60 +6,63 @@ displayed_sidebar: default
...
+[Composer](https://github.com/mosaicml/composer) is a library for training neural networks better, faster, and cheaper. It contains many state-of-the-art methods for accelerating neural network training and improving generalization, along with an optional [Trainer](https://docs.mosaicml.com/projects/composer/en/stable/trainer/using_the_trainer.html) API that makes _composing_ many different enhancements easy.
 
-W&B provides a lightweight wrapper for logging your ML experiments. But you don't need to combine the two yourself: Weights & Biases is incorporated directly into the Composer library via the [WandBLogger](https://docs.mosaicml.com/en/latest/api\_reference/composer.loggers.wandb\_logger.html#composer-loggers-wandb-logger).
+W&B provides a lightweight wrapper for logging your ML experiments. But you don't need to combine the two yourself: Weights & Biases is incorporated directly into the Composer library via the [WandBLogger](https://docs.mosaicml.com/projects/composer/en/stable/trainer/file_uploading.html#weights-biases-artifacts).
```
Comment (ngrayluna) [inline docs/guides/integrations/other/composer.md]: Replace "Weights & Biases" with "W&B"

**Comment by morganmcg1** [comment]
Most linter comments addressed, I ignored the "use parenthesis judiciously" comment as I feel the parenthesis around the param arugments was fine?

**Pull request merged**